application/src/main/java/org/togetherjava/tjbot/features/chatgpt/ChatGptService.java

```diff
@@ -96,12 +97,51 @@ public Optional<String> ask(String question, String context) {
             return Optional.empty();
         }
 
+        String instructions = "KEEP IT CONCISE, NOT MORE THAN 280 WORDS";
+        String questionWithContext = "context: Category %s on a Java Q&A discord server. %s %s"
+            .formatted(context, instructions, question);
+
+        return getMessageResponse(questionWithContext);
+    }
+
+    /**
+     * Provide ChatGPT with code to format.
+     *
+     * @param code the code to be formatted by ChatGPT. If code exceeds {@value MAX_CODE_LENGTH}
+     *        characters then this method returns an empty {@link Optional}
+     * @return an optional response from ChatGPT as a String
+     */
+    public Optional<String> formatCode(CharSequence code) {
+        if (isDisabled || code.length() > MAX_CODE_LENGTH) {
+            return Optional.empty();
+        }
+
+        String payload = String.format(
+                """
+                                If you happen to find any code in the container below, FORMAT \
+                                IT regardless of the programming language you find. MAKE IT HUMANLY READABLE. \
+                                Output with no introduction, no explanation, no ``` stuff, only code. Double \
+                                check that your response is correct. The code provided might not be readable.
+
+                                --- BEGIN CODE ---
+                                %s
+                                --- END CODE ---
+                        """,
+                code);
+
+        return getMessageResponse(payload);
+    }
+
+    /**
+     * Prompt ChatGPT with a message and get its response.
+     *
+     * @param message the message to send to ChatGPT
+     * @return an optional response from ChatGPT as a String
+     */
+    private Optional<String> getMessageResponse(String message) {
         try {
-            String instructions = "KEEP IT CONCISE, NOT MORE THAN 280 WORDS";
-            String questionWithContext = "context: Category %s on a Java Q&A discord server. %s %s"
-                .formatted(context, instructions, question);
-            ChatMessage chatMessage = new ChatMessage(ChatMessageRole.USER.value(),
-                    Objects.requireNonNull(questionWithContext));
+            ChatMessage chatMessage =
+                    new ChatMessage(ChatMessageRole.USER.value(), Objects.requireNonNull(message));
```

Comment:
make sure to add context in help thread chat gpt feature so that doesn't break

Reply:
`ChatGptService` contains ask method which also has context and is used for both help threads and `/chatgpt` feature

Reply:
yes, and this pr is removing that to make it more generic. So the context should be moved to the using sites

Reply:
![image](https://github.com/Together-Java/TJ-Bot/assets/61616007/b89a6af9-db0c-43f1-949d-7594ad751439)

I'm not sure what you mean by removing it, i still see context here in local copy of this PR.